JavaScript/JavaScriptSDK/Context/PageVisitTimeManager.ts

```diff
@@ -0,0 +1,96 @@
+﻿
+module Microsoft.ApplicationInsights.Context {
+    "use strict";
+
+    /**
+     * Used to track page visit durations
+     */
+    export class PageVisitTimeManager {
+
+        private prevPageVisitDataKeyName = "prevPageVisitData";
+
+        /**
+         * Stops timing of current page (if exists) and starts timing for duration of visit to pageName
+         * @param pageName Name of page to begin timing visit duration
+         * @returns {PageVisitData} Page visit data (including duration) of pageName from last call to start or restart, if exists. Null if not. 
+         */
+        restartPageVisitTimer(pageName: string, pageUrl: string) {
+
+            var prevPageVisitData = this.stopPageVisitTimer();
+            this.startPageVisitTimer(pageName, pageUrl);
+
+            return prevPageVisitData;
+        }
+
+        /**
+         * Starts timing visit duration of pageName
+         * @param pageName 
+         * @returns {} 
+         */
+        startPageVisitTimer(pageName: string, pageUrl: string) {
+            if (this.checkSessionStorageSupported()) {
+                if (window.sessionStorage.getItem(this.prevPageVisitDataKeyName) != null) {
+                    throw new Error("Cannot call startPageVisit consecutively without first calling stopPageVisit");
```

Comment:
What happens when you throw from here? Sorry for not having concrete suggestions - please see the similar code like PageViewPerformance.ts when we handle absense of some API. 
Please make sure that
1) this code doesn't affect other functionality
2) we get meaningful telemetry about not sending session events
3) if customer sees errors then its actionable, otherwise send it as not actionable (see Logging.throwXXX() methods)